core/src/stored/append.cc

```diff
@@ -110,6 +110,24 @@ static bool SaveFullyProcessedFilesAttributes(
   return false;
 }
 
+bool DeleteNullJobmediaRecords(JobControlRecord* jcr)
```

Comment:
While the file name fits, the contents do not.  Basically everything in `askdir.cc` is about dcrs.  I do not think this function should become a member function of a dcr.
If I move it into `askdir.cc` should I also create `askdir.h` or should I just use
```
extern bool DeleteNullJobmediaRecords(JobControlRecord* jcr);
```
?

Reply:
I see what you mean. I guess there is no harm then to just keep the `deletenullrecords` function within `append.cc` for now. 
if you feel it right, you can rename askdir to something else to reflect that it's the `dcr` asking for things, and maybe another file for free director requests (I guess the null records is the only one for now, unless there are others hidden somewhere). 
either way it's fine for me

Reply:
I moved it into `askdir.cc` but and used a prototype to declare it in append.cc.  If we find other such functions i think it will make sense to create a separate header for them.

Reply:
If you can avoid it, please don't put non-static function prototypes in .cc files.